packages/server/lib/modes/record.js

```diff
@@ -108,145 +109,300 @@ const getSpecRelativePath = (spec) => {
   return _.get(spec, 'relative', null)
 }
 
-const uploadArtifacts = (options = {}) => {
-  const { protocolManager, video, screenshots, videoUploadUrl, captureUploadUrl, screenshotUploadUrls, quiet } = options
-
-  const uploads = []
-  const uploadReport = {
-    protocol: undefined,
-    screenshots: [],
-    video: undefined,
+/*
+artifacts : [
+  {
+    reportKey: 'protocol' | 'screenshots' | 'video',
+    uploadUrl: string,
+    filePath?: string,
+    url: string,
+    fileSize?: number | bigint,
+    payload?: Buffer,
+    message?: string,
+  }
+]
+
+returns:
+[
+  {
+    success: boolean,
+    error?: string,
+    url: artifact.uploadUrl,
+    pathToFile: artifact.filePath,
+    fileSize: artifact.fileSize,
+    key: artifact.reportKey,
+  },
+  ...
+]
+*/
+
+const uploadArtifactBatch = async (artifacts, protocolManager, quiet) => {
+  const priority = {
```

Comment:
Why 0, 1, and 3 instead of 0, 1, and 2?

Reply:
used to have a 4th entry - thanks for catching this